usaspending_api/database_scripts/job_archive/backfill_faba_distinct_award_key.py

```diff
@@ -0,0 +1,111 @@
+"""
+Jira Ticket Number(s): DEV-5943
+
+    Create a new field which can determine distinct awards in FABA
+
+Expected CLI:
+
+    $ python3 usaspending_api/database_scripts/job_archive/backfill_faba_distinct_award_key.py
+
+Purpose:
+
+    Generates distinct_award_key for all FinancialAccountsByAwards records.
+
+    SINGLE PROCESS VERSION
+"""
+import math
+import psycopg2
+import time
+
+from os import environ
+
+
+# DEFINE THIS ENVIRONMENT VARIABLE BEFORE RUNNING!
+# Obvs, this is the connection string to the database.
+CONNECTION_STRING = environ["DATABASE_URL"]
+
+
+CHUNK_SIZE = 50000
+TOTAL_UPDATES = 0
+
+SQLS = [
+    """
+update  financial_accounts_by_awards
+set     distinct_award_key = concat(piid, '|', parent_award_id, '|', fain, '|', uri)
+where   financial_accounts_by_awards_id between {minid} and {maxid}
+    and distinct_award_key is distinct from concat(piid, '|', parent_award_id, '|', fain, '|', uri)
```

Comment:
I think you need to standardize on casing here, and e.g. uppercase everything, since Broker DABS submitters may use mixed casing for things like FAIN and URI on different submissions.

Reply:
Done